### PR TITLE
[SYCL] Test for set_final_data(..) method

### PR DESCRIPTION
**Pull request by maximdimakov**
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

File: SYCL/Basic/buffer/buffer-set_final_data.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      cgh.parallel_for<class FillBuffer>(
+          r, [=](cl::sycl::id<1> WIid) { Accessor[WIid] = false; });
+    });
+  } // Data is copied back
+
+  for (size_t i = 0; i < size; i++) {
+    if (data_vector[i] != true)
+      assert(false);
+  }
+}
```
Comment (s-kanaev) [inline SYCL/Basic/buffer/buffer-set_final_data.cpp]: Lacks `return 0;`

**Review by s-kanaev — COMMENTED** [review]
Looks good.
Though, should other types be tested here as well to ensure that there is no failure?

**Comment by s-kanaev** [comment]
Also, there're tests for `set_final_data` in https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Basic/buffer/buffer.cpp.
I believe, this test can be added there.

File: SYCL/Basic/buffer/buffer-set_final_data.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Queue.submit([&](cl::sycl::handler &cgh) {
+      auto Accessor = buf_shrd.get_access<cl::sycl::access::mode::write>(cgh);
+      cgh.parallel_for<class FillBuffer>(
+          r, [=](cl::sycl::id<1> WIid) { Accessor[WIid] = false; });
+    });
+  } // Data is copied back
+
+  for (size_t i = 0; i < size; i++) {
+    if (data_vector[i] != true)
+      assert(false);
```
Comment (vladimirlaz) [inline SYCL/Basic/buffer/buffer-set_final_data.cpp]: can you write some message and return 1 to be more user friendly?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

**Comment by maximdimakov** [comment]
@s-kanaev The test was moved to the buffer.cpp. It was extracted for int and double types. 
@vladimirlaz  Assert message was added

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**